app/controllers/assessment/autograde.rb

```diff
@@ -448,23 +456,17 @@ def saveAutograde(submissions, feedback)
       @assessment.problems.each do |p|
         submissions.each do |submission|
           score = submission.scores.find_or_initialize_by(problem_id: p.id)
-          score.score = 0
-          score.feedback = feedback_str
-          score.released = true
-          score.grader_id = 0
-          score.save!
+          if score.new_record? then # don't overwrite scores
+            score.score = 0
+            score.feedback = feedback_str
+            score.released = true
+            score.grader_id = 0
+            score.save!
+          end
```

Comment:
It might be a new_record if the autograder did not set some problem scores (like style scores).  But if a TA sets a score for a problem, it will not be overwritten.